docker/test-server/setup.sh

```diff
@@ -0,0 +1,26 @@
+#! /bin/bash
+
+set -ex
+
+apk add libpq postgresql-dev --virtual .build-deps coreutils gcc libc-dev libffi-dev make
+
+mkdir -p /app /data/static /data/media
+
+cd /app
+
+tar -xvf /context/archive.tar
```

Comment:
You don't need to archive anything. Here's how the Dockerfile should look like (similar to what victorrds surely has):
```
ENV PIP_DISABLE_PIP_VERSION_CHECK=1
ENV PIP_NO_CACHE_DIR=1

COPY requirements.txt /app/
RUN pip install --no-cache-dir -r /app/requirements.txt

COPY . /app

WORKDIR /app
```
Also, please put the contents of this file directly in the Dockerfile rather than in a different file. It's more standard, you can do multline like this:
```
RUN set -ex ;\
        apt-get update ;\
        apt-get install -y wait-for-it
```

Reply:
The downside of this inlining is that commenting and whitespace are more difficult.  But in this case there's not a lot of comments, so I'll do as you request.  And, good point about moving the `requirements.txt` and `pip install` up in the Docker steps.

Note that @victor-rds's docker image is AGPL-licensed, so I did my best effort to make a cleanroom implementation here (despite having looked at that repo for inspiration) rather than incidentally license etesync as AGPL.

Reply:
The server is AGPL, and anyhow this is the testing build script rather than the main server code and they can be differently licensed. (to my understanding).

Comment and whitespacing: I just split chunks of related code to different RUNs and put comments in between.

Reply:
Ah, I had looked at the clients' licenses.  That's good :)